README.md

```diff
@@ -1,37 +1,80 @@
-Boulder - An ACME CA
-====================
-
-This is an implementation of an ACME-based CA. The [ACME protocol](https://github.com/ietf-wg-acme/acme/) allows the CA to automatically verify that an applicant for a certificate actually controls an identifier, and allows domain holders to issue and revoke certificates for their domains.
+# Boulder - An ACME CA
 
 [![Build Status](https://travis-ci.org/letsencrypt/boulder.svg)](https://travis-ci.org/letsencrypt/boulder)
 [![Coverage Status](https://coveralls.io/repos/letsencrypt/boulder/badge.svg)](https://coveralls.io/r/letsencrypt/boulder)
 
-Development Setup
-------
+This is an implementation of an ACME-based CA. The [ACME protocol](https://github.com/ietf-wg-acme/acme/) allows the CA to automatically verify that an applicant for a certificate actually controls an identifier, and allows domain holders to issue and revoke certificates for their domains. Boulder is the software that runs [Let's Encrypt](https://letsencrypt.org).
+
+## Contents
+
+* [Overview](#overview)
+* [Setting up Boulder](#setting-up-boulder)
+  * [Development](#development)
+  * [Working with Certbot](#working-with-certbot)
+  * [Working with another ACME Client](#working-with-another-acme-client)
+  * [Production](#production)
+* [Contributing](#contributing)
+* [License](#license)
+
+## Overview
+
+Boulder is divided into the following main components:
+
+1. Web Front Ends (one per API version)
+2. Registration Authority
+3. Validation Authority
+4. Certificate Authority
+5. Storage Authority
+6. Publisher
+7. OCSP Updater
+8. OCSP Responder
+
+This component model lets us separate the function of the CA by security context.  The Web Front End, Validation Authority, and Publisher need access to the Internet, which puts them at greater risk of compromise.  The Registration Authority can live without Internet connectivity, but still needs to talk to the Web Front End and Validation Authority.  The Certificate Authority need only receive instructions from the Registration Authority. All components talk to the SA for storage, so lines indicating SA RPCs are not shown here.
```

Comment:
*"most lines indicating SA RPCs are not shown" -- since the RA->SA line is in fact shown.